cg/services/order_validation_service/rules/case_sample/utils.py

```diff
@@ -1,19 +1,48 @@
+from collections import Counter
+from cg.constants.constants import PrepCategory
+from cg.constants.sample_sources import SourceType
+from cg.constants.subject import Sex
 from cg.models.orders.sample_base import ContainerEnum
-from cg.services.order_validation_service.workflows.tomte.models.sample import (
-    TomteSample,
+from cg.services.order_validation_service.constants import MAXIMUM_VOLUME, MINIMUM_VOLUME
+from cg.services.order_validation_service.errors.case_errors import RepeatedCaseNameError
+from cg.services.order_validation_service.errors.case_sample_errors import (
+    FatherNotInCaseError,
+    InvalidConcentrationIfSkipRCError,
+    InvalidFatherSexError,
+    InvalidMotherSexError,
+    MotherNotInCaseError,
+    OccupiedWellError,
+    SampleNameRepeatedError,
+    SubjectIdSameAsCaseNameError,
 )
+from cg.services.order_validation_service.models.aliases import (
+    CaseContainingRelatives,
+    SampleWithRelatives,
+)
+from cg.services.order_validation_service.models.order_with_cases import OrderWithCases
+from cg.services.order_validation_service.models.sample import Sample
+from cg.store.models import Application
 from cg.store.store import Store
 
 
-def is_concentration_missing(sample: TomteSample) -> bool:
+def is_application_not_compatible(
+    allowed_prep_categories: list[PrepCategory],
+    application_tag: str,
+    store: Store,
+) -> bool:
+    application: Application = store.get_application_by_tag(application_tag)
+    return application and (application.prep_category not in allowed_prep_categories)
```

Comment:
Why is this added here? We have it in order_validation_service/rules/utils.py already

Reply:
https://github.com/Clinical-Genomics/cg/pull/3625/commits/62c1a54290cd0ff16666334e1685b66f4fa34cc7